build-tools/src/main/java/org/elasticsearch/gradle/plugin/PluginBuildPlugin.java

```diff
@@ -208,7 +208,7 @@ public void execute(Task task) {
              */
             zip.from(new Closure<Object>(null, null) {
                 public Object doCall(Object it) {
-                    return project.getPlugins().hasPlugin(ShadowPlugin.class)
+                    return project.getPluginManager().hasPlugin("com.github.johnrengelman.shadow")
```

Comment:
out of curiosity, why do you prefer this? I somehow still prefer the `project.getPlugins().hasPlugin(ShadowPlugin.class)` syntax as it makes refactorings and lookups much easier

Reply:
I'd prefer that _syntax_ as well, but then that requires us to bring in the Shadow plugin on the compile classpath, thus have it be a dependency for plugin projects. This also means they have to add the gradle plugin portal as a repository to their builds. This just makes things simpler for consumers.